### PR TITLE
Use PSR-15

### PR DESCRIPTION
**Pull request by jakejohns**

### CODE REVIEW

**Comment by franzliedke** [comment]
This should probably go to a new branch for the next major release, due to the breaking changes.

As an alternative (which sounds preferable to me), we could just offer a new middleware class instead of changing the old one.

**Comment by aidantwoods** [comment]
@jakejohns would you mind re-targeting this PR to the `2.0` branch? (I'm conscious that `1.0` still doesn't exist yet – I should probably tag and release that! 😬)

Since the PSR7 adapter will be a recommended thing to move to when SecureHeaders v2.1 is released (and deprecates the internal adapter) it would be good to keep it compatible with at least the same versions of PHP that the main library will be.
That said, there are plans for the next major version of SecureHeaders to pull the plug on PHP 5, so I see no reason not to release the 2.0 version of this library at the same time as the 1.0 version so that people who don't need PHP 5 can just move right up straight away.

File: composer.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "type": "library",
     "require": {
         "aidantwoods/secureheaders": "^2.0",
-        "psr/http-message": "^1.0"
```
Comment (aidantwoods) [inline composer.json:11]: Could you add `"php": ">=7.0"`?

File: tests/ApplySecureHeadersTest.php [hunk 2]
```diff
@@ -7,24 +7,27 @@
...
         $headers  = new SecureHeaders;
         $headers->errorReporting(false);
         $handler  = new PsrHttpAdapter\ApplySecureHeaders($headers);
 
-        $next = function ($requestIn, $responseIn) use ($request, $response) {
-            $this->assertSame($request, $requestIn);
-            $this->assertSame($response, $responseIn);
-            return $responseIn;
+        $next = new class implements Handler {
+            public function handle(Request $request) : REsponseInterface
```
Comment (aidantwoods) [inline tests/ApplySecureHeadersTest.php]: s/REsponseInterface/ResponseInterface/ ;-)

File: composer.json (hunk 1 above)
Comment (aidantwoods) [reply composer.json:11]: Also could you remove the <7.0 tests from `.travis.yml` and `.appveyor.yml`

**Review by aidantwoods — DISMISSED** [review]
Thanks for this! 😃 Looks like PSR-15 was moved to "accepted" just 5 days ago, so I wasn't even aware it had happened!

**Comment by jakejohns** [comment]
I believe this is updated as per your request.
Sorry about the delay!

**Comment by franzliedke** [comment]
Looking good so far. It needs an update to the README as well.

**Comment by jakejohns** [comment]
I pushed a first attempt here. Let me know if you have thought on how to explain.

**Review by franzliedke — APPROVED** [review]

**Pull request merged**